addon/utils/locale.js

```diff
@@ -13,6 +15,10 @@ export default class Locale {
   //  3. Default `rtl` to `false`
   //  4. Ensure `pluralForm` is defined
   constructor(id, container) {
+    if (initializedLocales[id]) {
+      return initializedLocales[id];
+    }
```

Comment:
Alright. For now I'll proceed with using this branch in my app, since it seems stable and I haven't noticed any performance hits (I have a small number of translations, and the cross-locale lookup is only performed once on page load). I will keep an eye on that task.